src/proxy/topic/TopicProxy.ts

```diff
@@ -13,22 +12,23 @@ export class TopicProxy<E> extends PartitionSpecificProxy implements ITopic<E> {
 
     publish(message: E): Promise<void> {
         assertNotNull(message);
-        const data = this.toData(message);
-        const request = TopicPublishCodec.encodeRequest(this.name, data);
         return this.encodeInvoke<void>(TopicAddMessageListenerCodec,
                                        TopicAddMessageListenerCodec.decodeResponse);
```

Comment:
You should use publish codec

Reply:
I recomment you to check other partitionspecificproxies and how they use encodeInvoke